src/System.Net.Http/src/System/Net/Http/Managed/HttpConnectionHandler.cs

```diff
@@ -91,7 +91,7 @@ protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessa
 
             if (HttpUtilities.IsSupportedSecureScheme(uri.Scheme))
             {
-                SslStream sslStream = await EstablishSslConnection(uri.IdnHost, request, stream).ConfigureAwait(false);
+                SslStream sslStream = await EstablishSslConnection(request.Headers.Host ?? uri.IdnHost, request, stream).ConfigureAwait(false);
```

Comment:
Should you remove the optional port first? I assume EstablishSslConnection isn't expecting that.

Reply:
Sure, fixed.